packages/functions_client/lib/src/functions_client.dart

```diff
@@ -84,12 +90,36 @@ class FunctionsClient {
       if (headers != null) ...headers
     };
 
-    final request = http.Request(method.name, uri);
+    final http.BaseRequest request;
+    if (files != null) {
+      assert(
+        body == null || body is Map<String, String>,
+        'body must be of type Map',
+      );
+      final fields = body as Map<String, String>?;
+
+      request = http.MultipartRequest(method.name, uri)
+        ..fields.addAll(fields ?? {})
+        ..files.addAll(files);
+    } else {
+      final bodyRequest = http.Request(method.name, uri);
+
+      final String? bodyStr;
+      if (body == null) {
+        bodyStr = null;
+      } else if (body is String) {
+        bodyStr = body;
+      } else {
+        bodyStr = await _isolate.encode(body);
```

Comment:
It'd be good to add tests for `body` with different types, like `int`, `double`, `Map` and `List`.